newton/_src/utils/heightfield.py

```
@@ -347,3 +347,192 @@ def heightfield_vs_convex_midphase(
+    z_range = hfd.max_z - hfd.min_z
+    base = hfd.data_offset
+
+    # Starting cell from entry point.
+    entry = ro + rd * t_enter
+    col = wp.int32(wp.floor((entry[0] + hfd.hx) / dx))
+    row = wp.int32(wp.floor((entry[1] + hfd.hy) / dy))
+    col = wp.clamp(col, 0, hfd.ncol - 2)
+    row = wp.clamp(row, 0, hfd.nrow - 2)
+
+    # DDA deltas and first-boundary parameters per XY axis.
+    step_col = int(0)
```

Ruff (0.15.10)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)